tools/pylib/boutdata/processor_rearrange.py

```diff
@@ -0,0 +1,92 @@
+"""Routines for redistributing files over different numbers of processors"""
+
+from math import sqrt
+
+
+def get_processor_layout(boutfile, has_t_dimension=True, mxg=2, myg=2):
+    """
+    Given a BOUT.restart.* or BOUT.dmp.* file (as a DataFile object), return the processor layout for its data
+    """
+    nxpe = boutfile.read('NXPE')
+    nype = boutfile.read("NYPE")
+    npes = nxpe * nype
+
+    # Get list of variables
+    var_list = boutfile.list()
+    if len(var_list) == 0:
+        raise ValueError("ERROR: No data found")
+
+    mxsub = 0
+    mysub = 0
+    mz = 0
+    if has_t_dimension:
+        maxdims = 4
+    else:
+        maxdims = 3
+    for v in var_list:
+        if boutfile.ndims(v) == maxdims:
+            s = boutfile.size(v)
+            mxsub = s[maxdims - 3] - 2 * mxg
+            if mxsub < 0:
+                if s[maxdims - 3] == 1:
+                    mxsub = 1
+                    mxg = 0
+                elif s[maxdims - 3] == 3:
+                    mxsub = 1
+                    mxg = 1
+                else:
+                    print("Number of x points is wrong?")
+                    return False
+
+            mysub = s[maxdims - 2] - 2 * myg
+            if mysub < 0:
+                if s[maxdims - 2] == 1:
+                    mysub = 1
+                    myg = 0
+                elif s[maxdims - 2] == 3:
+                    mysub = 1
+                    myg = 1
+                else:
+                    print("Number of y points is wrong?")
+                    return False
+
+            mz = s[maxdims - 1]
+            break
+
+    # Calculate total size of the grid
+    nx = mxsub * nxpe
+    ny = mysub * nype
+
+    return nxpe, nype, npes, mxsub, mysub, nx, ny, mz, mxg, myg
```

Comment:
Given the number of return values, it might be better to wrap these up in a `dict` or `namedtuple`